### DISCUSSION TITLE
Large File Uploads to Documents in Open WebUI Workspace Fail on GCE

### DISCUSSION THREAD

**Discussion by JeJoeWa** [open-webui/open-webui#3958] [General]
Description
Bug Summary:
When attempting to upload "large files" (over 5 MB) using the Open WebUI interface, the upload fails silently. Smaller files upload and process without issues.

**Steps to Reproduce:**
Set up Open WebUI following the installation guide for Installing Open WebUI with Bundled Ollama Support

Attempt to upload a small file (e.g., under 5 MB) through the Open WebUI interface and Documents (RAG).
Observe that the file uploads successfully and is processed.

Attempt to upload a large file through the Open WebUI interface.
Observe that the upload fails silently without any errors in the logs.

Expected Behavior:
Large files should upload successfully and be processed similarly to smaller files.

Actual Behavior:
Large files fail to upload, and there is no indication in the logs that the POST request to /rag/api/v1/process/doc is being made.

**Environment**
Open WebUI Version: 0.3.10
Operating System: Ubuntu 20.04 on Google Compute Engine (GCE)
Browser (if applicable): Chrome 103.0

**Reproduction Details**
- I have read and followed all the instructions provided in the README.md.
- I am on the latest version of both Open WebUI and Ollama.
- I have included the Docker container logs.
- No relevant errors in the browser console.

**Docker Container Logs:**

Logs for successful upload of a small file:
```
INFO  [apps.rag.main] store_data_in_vector_db [Document(metadata={'source': '/app/backend/data/uploads/example.pdf', ... }]
INFO  [apps.rag.main] store_docs_in_vector_db [Document(metadata={'source': '/app/backend/data/uploads/example.pdf', ... }]
INFO:     127.0.0.1:46985 - "POST /rag/api/v1/process/doc HTTP/1.1" 200 OK
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
```

Logs for failed upload of a large file:
```
INFO  [apps.rag.main] store_data_in_vector_db [Document(metadata={'source': '/app/backend/data/uploads/large-file.pdf', ... }]
INFO  [apps.rag.main] store_docs_in_vector_db [Document(metadata={'source': '/app/backend/data/uploads/large-file.pdf', ... }]
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
```

**Installation Method**
Installed using Docker on Google Compute Engine (GCE).

Additional Information
Verified that there are no GCE firewall rules or network settings limiting the upload size.
No explicit error messages in the logs indicating why the upload fails.
Smaller files upload and process without issues, indicating the general setup is correct.

After further testing, I discovered additional details about the issue with uploading large files:
Local Environment Success:
When running the Docker container locally and accessing the Open WebUI interface directly through localhost:3000, large file uploads (over 100 MB) succeed without any issues.

Cloudflare Tunnel:
When exposing the Docker container through a Cloudflare tunnel and attempting to upload large files, the upload fails similarly to when running on Google Compute Engine (GCE).

**Comment by raphael2692** [comment]
Same issue using docker desktop on Windows and openAI (docker run -d -p 3000:8080 -e OPENAI_API_KEY=sk-... -v open-webui:/app/backend/data --name open-webui --restart always [ghcr.io/open-webui/open-webui:main](http://ghcr.io/open-webui/open-webui:main)

**Reply by derm1ch1** [reply]
I am facing this issue as well. Is there any difference if you use command line tools instead of the webui?

**Comment by bekmuradov** [comment]
Hi @JeJoeWa ,

I encountered a similar issue when uploading large files over HTTPS, but not HTTP. After setting up a custom domain, I noticed that large file uploads would fail over HTTPS. If your situation is similar, it's likely due to configuration settings in Nginx or Apache, depending on which you're using.

### Steps to Resolve the Issue:

1. **Check the Network Tab:**
   - Open your browser's developer tools and check the Network tab.
   - If you see an error like `413 Request Entity Too Large nginx/1.18.0`, this indicates that Nginx's default `client_max_body_size` is too small for your upload.

2. **Update Nginx Configuration:**
   - SSH to your GCE
   - Open your Nginx configuration file:
     ```bash
     sudo nano /etc/nginx/sites-available/default
     ```
   - Add or update the `client_max_body_size` directive to allow larger uploads:
     ```nginx
     http {
         client_max_body_size 100M;
         ...
     }
     ```
   - Reload Nginx to apply the changes:
     ```bash
     sudo systemctl reload nginx
     ```

3. **If Using Apache:**
   - For Apache, you would need to adjust the `LimitRequestBody` directive:
     ```apache
     <Directory "/var/www/html">
         LimitRequestBody 52428800
     </Directory>
     ```
   - Restart Apache to apply the changes:
     ```bash
     sudo systemctl restart apache2
     ```

Try these steps and see if it resolves the issue.

**Reply by madsit83** [reply]
Good catch, adapting the nginx config fixxed the issue for me, default setting seems to be 1mb max file size.

**Comment by TristanB1994** [comment]
This is still an issue, not using any webserver infront of openweb

**Reply by rminchev1** [reply]
Any luck yet with this?

**Reply by ablerman** [reply]
@TristanB1994 We saw something similar on AWS. The LoadBalancers timeout after 30 seconds, so if the processing takes more than 30 seconds, the fileupload fails. Are you able to test that theory? Are the uploads failing when the connection times out?

I was able to temporarily fix the issue by increasing the timeout to 10 minutes. In a multiuser environment, this probably isn't a permanent solution.

**Reply by rminchev1** [reply]
Hey guys, it was aws alb setting causing the timeout in my case, it was Idle time that had to be increased, then everything worked great. The default was 60 sec, I bumped it to 300 sec.

**Reply by TristanB1994** [reply]
@ablerman for me this is 100% local so no AWS LB settings to blame!

**Reply by computerdane** [reply]
+1 on the ALB timeout, fixed my issue

**Comment by Omhra-Anh** [comment]
I encountered this issue and solved by increasing the allowed upload size, first by adding a line in the nginx  reverse proxy because it limits to 1 mb by default.  go to "/etc/nginx/sites-available/default (or whatever your config file is named) and add the following line within in the server block, bellow the SSL configuration:

#Client max body size
client_max_body_size 300M;

And then after lots of work along with the A.I. the rest was solved. I am running in docker (not using docker desktop) but the process should be very similar.

I have tested with .pdf up to 20mb, and it works. Be prepared for some intensive CPU work. It took my 2950X Threadripper about 8 minutes to crunch it after about 8 minutes of "upload" ...all cores running high.
After placing the knowledge in the model, a query successfully produced a summary of the book.

Always do a backup before attempting any fix.. good luck!!!

**This is the process:**

How to Increase the File Upload Size Limit in Open-WebUI.

Open-WebUI enforces a file upload size limit through an environment variable (RAG_FILE_MAX_SIZE) and its configuration files. If you need to increase this limit (e.g., to 300MB), follow these steps.

-----------------------------------------------
Step 1: Access the Open-WebUI Container
If Open-WebUI is running in a Docker container, first enter the container:
docker exec -it <container-name> /bin/bash
Replace <container-name> with the actual name of your Open-WebUI container. (most likely docker exec -it open-webui bash)
If Open-WebUI is not in a container, simply open a terminal.
________________________________________
Step 2: Modify the Configuration File
The file upload limit is defined in config.py, located inside the Open-WebUI backend.
1.	Open the file in a text editor:
nano /app/backend/open_webui/config.py
2.	Find the section that looks like this:
RAG_FILE_MAX_SIZE = PersistentConfig(
    "RAG_FILE_MAX_SIZE",
    "rag.file.max_size",
    (
        int(os.environ.get("RAG_FILE_MAX_SIZE"))
        if os.environ.get("RAG_FILE_MAX_SIZE")
        else None
    ),
)
3.	Modify it to set a default file upload size (e.g., 300MB):
RAG_FILE_MAX_SIZE = PersistentConfig(
    "RAG_FILE_MAX_SIZE",
    "rag.file.max_size",
    (
        int(os.environ.get("RAG_FILE_MAX_SIZE"))
        if os.environ.get("RAG_FILE_MAX_SIZE")
        else 300 * 1024 * 1024  # Default to 300MB
    ),
)
4.	Save the file:
o	Press Ctrl+X to exit.
o	Press Y to confirm.
o	Press Enter to save.
________________________________________
Step 3: Update the Environment Variable
The file size limit is also controlled by an environment variable. Update it with:
export RAG_FILE_MAX_SIZE=$((300 * 1024 * 1024))
To make this change permanent, modify the .env file:
nano /app/.env
Add or modify the following line:
RAG_FILE_MAX_SIZE=314572800
Save and exit (Ctrl+X, Y, Enter).
________________________________________
Step 4: Restart Open-WebUI
To apply the changes, restart Open-WebUI.
If Running in a Docker Container
docker restart <container-name>
If Running Natively
If Open-WebUI is running as a process, restart it manually:
1.	Find the process ID (PID):
ps aux | grep uvicorn
Look for a line like:
root         588  7.0 11.9 19959232 1946788 ?    Ssl  02:24  74:22 /usr/local/bin/python /usr/local/bin/uvicorn open_webui.main:app --host 0.0.0.0 --port 8080 --forwarded-allow-ips *
The first number (588) is the process ID (PID).
2.	Kill the Open-WebUI process:
kill -9 <PID>
Replace <PID> with the actual process ID.
3.	Restart Open-WebUI:
cd /app
./start.sh  # Use the actual startup command if different
________________________________________
Step 5: Test the New Upload Limit
Now, try uploading a larger file (e.g., 250MB) to verify that the limit is successfully increased.
________________________________________
Final Notes
•	This method sets a default file upload limit in case no environment variable is provided.
•	If you need to change the limit again, simply repeat Steps 2 and 3 with a new value.
•	Restarting Open-WebUI is necessary for the changes to take effect.

You're now ready to upload large files in Open-WebUI!

**Reply by 0xGitGuy** [reply]
Can't you do this via the UI under settings > document > general settings > Max upload size?

**Reply by freeload101** [reply]
Can confirm same https issue on Cloud Flare! Thanks this one had me stumped for a while! 

So try it http://localhost:8080/ first THEN start looking for gremlins 

![{F2A54093-7D80-4DF9-AE7E-A69DF2D4303A}](https://github.com/user-attachments/assets/c3dd753d-970b-48fc-a5a1-971f2742f4b7)

Cloudflare has a default HTTP response timeout of 100 seconds 
[9](https://community.cloudflare.com/t/default-timeout-limit-increase/240042)
[10](https://community.cloudflare.com/t/increase-cloudflare-max-request-duration/482873)
. If your origin server takes longer than this to respond to a request, Cloudflare will close the connection, which can result in a 524 timeout error 
[11](https://community.cloudflare.com/t/increase-timeout-execution-for-large-http-processes/427350)
[12](https://stackoverflow.com/questions/31051138/how-to-disable-cloudflare-custom-error-pages-including-error-504-gateway-time-o)
.

For users on Free, Pro, and Business plans, this 100-second timeout limit cannot be increased 
[10](https://community.cloudflare.com/t/increase-cloudflare-max-request-duration/482873)
[13](https://stackoverflow.com/questions/38536167/is-it-possible-to-increase-cloudflare-time-out)
.
Only Enterprise customers can increase the timeout limit, up to 6,000 seconds, by making a request via the Cloudflare API 
[11](https://community.cloudflare.com/t/increase-timeout-execution-for-large-http-processes/427350)
[14](https://community.cloudflare.com/t/i-need-to-change-the-default-cloudflare-timeout-from-100-seconds-to-300/719402)
.

**Comment by convernaticspeter** [comment]
Looks like this issue:
https://github.com/open-webui/open-webui/issues/12276

**Reply by rgaricano** [reply]
nop, this seem it was for upload size limits and that other one you reference looks more like some kind of connection timeout.

**Comment by Omhra-Anh** [comment]
Extending the timeouts worked for me. The only remaining challenge is the format and "locked" state of some pdf docs.
